app/models/pet.rb

```diff
@@ -9,11 +9,24 @@ def shelter_name
     shelter.name
   end
 
+  def set_adoptable_false
+    self.adoptable = false
+    self.save
+  end
+
+  def self.find_pet(id)
+    Pet.find(id)
+  end
+
   def self.adoptable
     where(adoptable: true)
   end
 
   def application_pet_approved(id)
     ApplicationPet.where(pet_id: self.id, application_id: id).pluck(:application_approved).first
   end
+
+  def already_approved?
+    adoptable ? false : true
```

Comment:
this is a mind bender lol. could we simplify it to just `!adoptable`? so it will return whatever the opposite of adoptable is?